authz/audit/audit_logging_test.go

```diff
@@ -289,12 +289,15 @@ func (s) TestAuditLogger(t *testing.T) {
 			defer cancel()
 
 			if _, err := client.UnaryCall(ctx, &testpb.SimpleRequest{}); status.Code(err) != test.wantUnaryCallCode {
-				t.Fatalf("Unexpected UnaryCall fail: got %v want %v", err, test.wantUnaryCallCode)
+				t.Errorf("Unexpected UnaryCall fail: got %v want %v", err, test.wantUnaryCallCode)
 			}
 			if _, err := client.UnaryCall(ctx, &testpb.SimpleRequest{}); status.Code(err) != test.wantUnaryCallCode {
-				t.Fatalf("Unexpected UnaryCall fail: got %v want %v", err, test.wantUnaryCallCode)
+				t.Errorf("Unexpected UnaryCall fail: got %v want %v", err, test.wantUnaryCallCode)
+			}
+			stream, err := client.StreamingInputCall(ctx)
+			if err != nil {
+				t.Errorf("StreamingInputCall failed:%v", err)
```

Comment:
This one needs to be `Fatal` because using the `stream` is illegal.

Reply:
I need some help here - go/go-style/decisions.md?cl=head#keep-going suggests to prefer `Error` over `Fatal`

>to print out all of the failed checks in a single run

So for this test even if stream is not usable I can still proceed with unary calls to get the test results for it.
However the same doc suggests

>Calling t.Fatal is primarily useful for reporting an unexpected error condition

and unusable stream is definitely something we don't expect. 
WDYT?

Reply:
True, `Error` is nicer, _generally_, but this is a precondition for (some of) the rest of the test.  The "ideal" would be to skip just the streaming calls and the checks that depend on them happening, but that really clutters the code and isn't worth it _at all_.  Just `Fatal` since a precondition failed and keep things simple.

Reply:
Ok, sounds good - I changed `Errorf` to be used for

- `cmp.Diff` cases
- errors which `status.Code` is not expected